File: src/326PowerOfThree/326PowerOfThree.cpp
#include <iostream>
using namespace std;

class Solution {
public:
  bool isPowerOfThree(int n) {
    int i = 1;
    while (i < n && i > 0) {
      cout << i << " ";
      i *= 3;
    }
    return i == n;
  }
};

int main() {
  Solution s;
  cout << s.isPowerOfThree(2147483647) << endl;
  return 0;
}

File: src/326PowerOfThree/326PowerOfThree.rs
impl Solution {
    pub fn is_power_of_three(n: i32) -> bool {
        if n <= 0 {return false;}
        if n == 1 {return true;}
        n%3==0 && Self::is_power_of_three(n/3)
    }
}

impl Solution {
    pub fn is_power_of_three(n: i32) -> bool {
        if n <= 0 {return false;}
        let mut n = n;
        while n%3==0 {
            n /= 3;
        }
        n==1
    }
}
